src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -1686,7 +1686,15 @@ private EventBuilder DefineEventNoLock(string name, EventAttributes attributes,
         }
```

Comment:
Also change `public TypeInfo? CreateTypeInfo()` to 
```
        [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
        public TypeInfo CreateTypeInfo()
        {
            Type? type = CreateTypeImpl();
            Debug.Assert(type != null);
            return type;
        }
```

Reply:
I updated this to mimic what was done for `CreateType()`, which amounts to a public API that returns a non-null `TypeInfo`.